api/src/main/java/org/openmrs/module/appointmentscheduling/task/MarkAppointmentsAsMissedTask.java

```diff
@@ -22,6 +22,8 @@ public void execute() {
                 Appointment.AppointmentStatus.getAppointmentsStatusByTypes(Arrays.asList(Appointment.AppointmentStatusType.SCHEDULED)))) {
             appointment.setStatus(Appointment.AppointmentStatus.MISSED);
             appointmentService.saveAppointment(appointment);
+
+            appointmentService.addNewStatusToHistory(appointment);
```

Comment:
Instead of having to add this line, could you just replace the above two lines that set the status and save with:

`appointmentService.changeAppointmentStatus(appointment, Appointment.AppointmentStatus.MISSED)`